crew

```diff
@@ -529,17 +529,31 @@ def download
 
   uri = URI.parse url
   filename = File.basename(uri.path)
+  
   if source
-    sha256sum = @pkg.source_sha256
+    if @pkg.source_sha512
+      csum_type = 512
+      csum = @pkg.source_sha512
+    elsif @pkg.source_sha256
+      csum_type = 256
+      csum = @pkg.source_sha256
+    end
+  elsif @pkg.binary_sha512
+    csum_type = 512
+    csum = @pkg.binary_sha512[@device[:architecture]]
+  elsif @pkg.binary_sha256
+    csum_type = 256
+    csum = @pkg.binary_sha256[@device[:architecture]]
   else
-    sha256sum = @pkg.binary_sha256[@device[:architecture]]
+    abort 'No checksum provided for this package, aborting...'.lightred
   end
```

Comment:
`case...when...else...end` would be more elegant.

Reply:
Do you mean `case source...else...end`?

Reply:
You can use `when` instead of `elsif`.

Reply:
Done